portalnet/src/overlay.rs

```diff
@@ -441,6 +441,39 @@ where
         }
     }
 
+    pub async fn lookup_node(&self, target: NodeId) -> Vec<Enr> {
+        if target == self.local_enr().node_id() {
+            return vec![self.local_enr()];
+        }
+        if self.discovery.connected_peers().contains(&target) {
+            let enr = self
+                .discovery
+                .find_enr(&target)
+                .expect("cannot find enr expected to exist in routing table");
+            return vec![enr];
+        }
```

Comment:
IIUC, we do not do the following because the node may not be in the connected state?

```suggestion
        if let Some(enr) = self.discovery.find_enr(&target) {
            return vec![enr];
        }
```

why not look directly at our own routing table?

```rust
self.kbuckets
  .write()
  .iter()
  .filter(|entry| entry.status.is_connected())
  .find(|entry| *entry.node.key.preimage() == target)
```